node.cpp

```diff
@@ -0,0 +1,49 @@
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include "node.h"
+
+#define MAX_CHILDREN 100
```

Comment:
`const MAX_CHILDREN = 100;` better than using the preprocessor, but I wouldn't use a hardcoded number for this. Use a vector.